### PR TITLE
test,lint: use circleci for CI tests and lint

### PR DESCRIPTION
**Pull request by cosnicolaou**
Use circleci for tests and linting. However, the lint tests do not need to pass for a PR to be merged. This will be changed in a subsequent PR when the lint tests pass.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          name: goimports
+          command: |
+            go get golang.org/x/tools/cmd/goimports
+            go install -x golang.org/x/tools/cmd/goimports
+      - go/save-cache
+      - run:
+          name: vdlpath
+          command: |
+            echo "export VDLPATH=$CIRCLE_WORKING_DIRECTORY" >> $BASH_ENV
+            source $BASH_ENV
```
Comment (razvanm) [inline .circleci/config.yml:22]: Is the `source $BASH_ENV` doing anything? As far as I can tell each step runs in a separate shell.

**Comment by cosnicolaou** [comment]
source $BASH_ENV is the key, it sets VDLPATH. The circleci runner appears
to parse the output and set job-level env vars.
Or at least that's what their docs led me to believe!

Cheers, Cos.

**Comment by razvanm** [comment]
The example I'm looking is the one from https://circleci.com/docs/2.0/env-vars/#example-configuration-of-environment-variables. In there I see a `run` step that only appends to the `$BASH_ENV` without sourcing it. A later step is using the appended variable without sourcing anything. So my understanding is that the `$BASH_ENV` is source in each `run` step.

**Comment by cosnicolaou** [comment]
it may well work without the source step... it could just be the example I
was looking at used in that step or my first attempt at doing so did.

Cheers, Cos.

**Pull request merged**